src/fmu/tools/utilities/sample_attributes.py

```diff
@@ -0,0 +1,162 @@
+
+    points = xtgeo.points_from_surface(surface)
+    points.zname = newname if newname else Attrs.VALUES.value
+    df = points.get_dataframe()
+    if debug:
+        temp_dir = pathlib.Path(tempfile.gettempdir())
+        out = temp_dir / f"debug_points_{newname}.poi"
+        points.to_file(out)
+        print("Debug points written to:", out)
+
+    return df
+
+
+def sample_attributes_for_sim2seis(
+    grid: xtgeo.Grid,
+    attribute: xtgeo.RegularSurface,
+    attribute_error: xtgeo.RegularSurface | float = 0.05,
+    attribute_error_minimum: float | None = None,
+    region: xtgeo.GridProperty | None = None,
+    zone: xtgeo.GridProperty | None = None,
+    position: tuple[str, Literal["top", "center", "base"]] = ("", "center"),
+    **kwargs: Any,
+) -> pd.DataFrame:
+    """Sample attributes on grid resolution as poinst sets.
+
+    This usage is for setting attributes on grid resolution, e.g. a seismic attribute
+    (from a map) combined with a region parameter from the grid.
+
+    This is targeted to the "sim2seis" workflow in FMU.
+
+    Args:
+        grid: The grid to sample the attributes on.
+        attribute: The seismic (or custom) map/surface to sample the attribute from.
+        attribute_error: The error to apply to the attribute (optional).
+            Shall be absolute (positive) values. If the user wants to apply a polygons
+            with different error values, the user can ise surface-polygons functions
+            in xtgeo to achieve this.
+        attribute_error_minimum: The minimum error to apply to the attribute (optional).
+        region: The region parameter to sample from the grid (optional).
+        zone: The zone parameter to sample from the grid (optional).
+        position: The position to sample the attributes on the grid. This
+            shall be given as a tuple, as e.g. ("MyZone", "center") where the first
+            is zone name, and the second is position ("top", "center", "base") in
+            that zone. Default is ("", "center") which will take the middle layer
+            of the total grid. The zone name is case sensitive. If zone is not given,
+            the full grid interval will be applied to determine the layer.
+        **kwargs: Additional keywords (developer settings).
+
+    Returns:
+        pd.Dataframe: Points with the sampled attributes and attributes combined.
+    """
+    logger.info("Sampling attributes on grid resolution as points set.")
+
+    debug = kwargs.get("debug", False)
+
+    layer = _get_layer(position, grid, zone)
+    logger.debug("Layer index (base=1) to apply is: %s", layer)
+
+    template = xtgeo.surface_from_grid3d(
+        grid,
+        template="native",  # used to appromimate the native grid resolution
+        where=layer,
+        property="i",
+    )
+    # prepare the attribute and error to the grid
+    attribute_sampled = template.copy()
+    attribute_error_sampled = template.copy()
```

Comment:
Do we need these copies here or could we just use the template object to save memory?

Reply:
Think we need it (since xtgeo do "inplace" operations, not returning the object)